[MacOS] Linker error when building from source

**Required Info:**

- Operating System:
  - macOS 10.15
- Installation type:
  - From source
- ROS Version
  - ROS2 Foxy
- Version or commit hash:
  - Latest of foxy-devel
  
#### Steps to reproduce issue

When trying to build `slam_toolbox` from source on macOS, you get the following linker error:

```
Undefined symbols for architecture x86_64:
  "void tf2::fromMsg<geometry_msgs::msg::Quaternion_<std::__1::allocator<void> >, tf2::Quaternion>(geometry_msgs::msg::Quaternion_<std::__1::allocator<void> > const&, tf2::Quaternion&)", referenced from:
      slam_toolbox::SlamToolboxPlugin::InitialPoseCallback(std::__1::shared_ptr<geometry_msgs::msg::PoseWithCovarianceStamped_<std::__1::allocator<void> > >) in slam_toolbox_rviz_plugin.cpp.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

This build was run on CI, and thus the full logs and build parameters can be viewed at https://github.com/nkalupahana/ros2-foxy-macos/runs/2699823378?check_suite_focus=true.

--- a/rviz_plugin/slam_toolbox_rviz_plugin.hpp
+++ b/rviz_plugin/slam_toolbox_rviz_plugin.hpp
@@ -41,7 +41,7 @@
 #include "rviz_common/panel.hpp"
 #include "slam_toolbox/toolbox_msgs.hpp"
 #include "geometry_msgs/msg/pose_with_covariance_stamped.hpp"
-
+#include "tf2_geometry_msgs/tf2_geometry_msgs.h"
 
 class QLineEdit;
 class QSpinBox;

